### PR TITLE
MKS H43 Fix bug

### PR DESCRIPTION
**Pull request by ghost**
### Description

In this PR, in order to fix the BUG of MKS H43,The repaired content mainly includes:

- MKS H43 compilation problem
- Leveling control
- Show repair

### Requirements

This PR is for H43 use.

### Benefits

1) Fix the error of SGEN_L when compiling, I added the following piece of code to the HAL.h file of LPC1768. If you do not add these codes, the user will get an error when compiling for the first time.

 \#if HAS_DGUS_LCD

  \#define SERIAL_GET_TX_BUFFER_FREE() MSerial0.available();

 \#endif

2) If MESH_BED_LEVELING is enabled, The first leveling will not return to zero, so I modified it and added a G28 execution.

3) Fix waiting for boot logo

4) Fix garbled characters in the Chinese interface

5) Fix the problem that input parameters are not saved to EEPROM.I think this is more necessary. After the user enters his own parameters, if the parameters cannot be saved, they will continue to input these parameters repeatedly, which makes the user feel very cumbersome, so these additional input parameters need to be written into the EEPROM , It’s not a lot, and it doesn’t take up much space.

### Related Issues

1) If the Z axis is walked in steps of 0.01, the coordinates will be wrong.The coordinate value does not accumulate by 0.01 every time, sometimes it accumulates to 0.02.Even if it is controlled by sending commands through the serial port.

### CODE REVIEW

File: Marlin/src/lcd/extui/lib/dgus/mks/DGUSDisplayDef.cpp [hunk 1]
```diff
@@ -567,13 +567,13 @@ const struct DGUS_VP_Variable ListOfVP[] PROGMEM = {
...
       VPHELPER(VP_UNLOAD_Filament, nullptr, &ScreenHandler.MKS_FilamentUnLoad, nullptr),
-      VPHELPER(VP_Filament_distance, &distanceFilament, &ScreenHandler.GetManualFilament, ScreenHandler.DGUSLCD_SendFloatAsIntValueToDisplay<0>),
+      VPHELPER(VP_Filament_distance, &distanceFilament, &ScreenHandler.GetManualFilament, ScreenHandler.DGUSLCD_SendWordValueToDisplay),
       VPHELPER(VP_Filament_speed, &FilamentSpeed, &ScreenHandler.GetManualFilamentSpeed, ScreenHandler.DGUSLCD_SendWordValueToDisplay),
     #endif
   #endif
 
   #if HOTENDS >= 2
-    VPHELPER(VP_T_E1_Is, &thermalManager.temp_hotend[1].celsius, nullptr, ScreenHandler.DGUSLCD_SendWordValueToDisplay),
+    VPHELPER(VP_T_E1_Is, &thermalManager.temp_hotend[1].celsius, nullptr, ScreenHandler.DGUSLCD_SendFloatAsLongValueToDisplay<0>),
```
Comment (thinkyhead) [inline Marlin/src/lcd/extui/lib/dgus/mks/DGUSDisplayDef.cpp]: That temperature there is a 16-bit signed integer, not a `float`.

File: Marlin/src/lcd/extui/lib/dgus/mks/DGUSDisplayDef.cpp [hunk 2]
```diff
@@ -567,13 +567,13 @@ const struct DGUS_VP_Variable ListOfVP[] PROGMEM = {
     #if ENABLED(DGUS_FILAMENT_LOADUNLOAD)
       VPHELPER(VP_LOAD_Filament, nullptr, &ScreenHandler.MKS_FilamentLoad, nullptr),
       VPHELPER(VP_UNLOAD_Filament, nullptr, &ScreenHandler.MKS_FilamentUnLoad, nullptr),
-      VPHELPER(VP_Filament_distance, &distanceFilament, &ScreenHandler.GetManualFilament, ScreenHandler.DGUSLCD_SendFloatAsIntValueToDisplay<0>),
+      VPHELPER(VP_Filament_distance, &distanceFilament, &ScreenHandler.GetManualFilament, ScreenHandler.DGUSLCD_SendWordValueToDisplay),
```
Comment (thinkyhead) [inline Marlin/src/lcd/extui/lib/dgus/mks/DGUSDisplayDef.cpp]: That `distanceFilament` variable is a `float`.

Comment (ghost) [reply Marlin/src/lcd/extui/lib/dgus/mks/DGUSDisplayDef.cpp]: Yes, I've changed it

**Comment by thinkyhead** [comment]
Thanks for the patch! I took the liberty to put the data into proper structures and arrays and to patch a bunch of other things, so there are now just 4 persistent data objects instead of many.

**Pull request merged**